builtin/v10/check.go

```diff
@@ -0,0 +1,367 @@
+			}
+			summary, msgs := market.CheckStateInvariants(&st, tree.Store, actor.Balance, priorEpoch)
+			acc.WithPrefix("market: ").AddAll(msgs)
+			marketSummary = summary
+		case actorCodes[manifest.PaychKey]:
+			var st paych.State
+			if err := tree.Store.Get(tree.Store.Context(), actor.Head, &st); err != nil {
+				return err
+			}
+			summary, msgs := paych.CheckStateInvariants(&st, tree.Store, actor.Balance)
+			acc.WithPrefix("paych: ").AddAll(msgs)
+			paychSummaries = append(paychSummaries, summary)
+		case actorCodes[manifest.MultisigKey]:
+			var st multisig.State
+			if err := tree.Store.Get(tree.Store.Context(), actor.Head, &st); err != nil {
+				return err
+			}
+			summary, msgs := multisig.CheckStateInvariants(&st, tree.Store)
+			acc.WithPrefix("multisig: ").AddAll(msgs)
+			multisigSummaries = append(multisigSummaries, summary)
+		case actorCodes[manifest.RewardKey]:
+			var st reward.State
+			if err := tree.Store.Get(tree.Store.Context(), actor.Head, &st); err != nil {
+				return err
+			}
+			summary, msgs := reward.CheckStateInvariants(&st, tree.Store, priorEpoch, actor.Balance)
+			acc.WithPrefix("reward: ").AddAll(msgs)
+			rewardSummary = summary
+		case actorCodes[manifest.VerifregKey]:
+			var st verifreg.State
+			if err := tree.Store.Get(tree.Store.Context(), actor.Head, &st); err != nil {
+				return err
+			}
+			summary, msgs := verifreg.CheckStateInvariants(&st, tree.Store, priorEpoch)
+			acc.WithPrefix("verifreg: ").AddAll(msgs)
+			verifregSummary = summary
+		case actorCodes[manifest.DatacapKey]:
+			var st datacap.State
+			if err := tree.Store.Get(tree.Store.Context(), actor.Head, &st); err != nil {
+				return err
+			}
+			summary, msgs := datacap.CheckStateInvariants(&st, tree.Store)
+			acc.WithPrefix("datacap: ").AddAll(msgs)
+			datacapSummary = summary
+		case actorCodes[manifest.EvmKey]:
+			var st evm.State
+			if err := tree.Store.Get(tree.Store.Context(), actor.Head, &st); err != nil {
+				return err
+			}
+			msgs := evm.CheckStateInvariants(&st, tree.Store)
+			acc.WithPrefix("evm: ").AddAll(msgs)
+		case actorCodes[manifest.PlaceholderKey]:
+			acc.Require(actor.Head == emptyObjectCid, "Placeholder actor head %v unequal to emptyObjectCid %v", actor.Head, emptyObjectCid)
+		case actorCodes[manifest.EthAccountKey]:
+			acc.Require(actor.Head == emptyObjectCid, "EthAccount actor head %v unequal to emptyObjectCid %v", actor.Head, emptyObjectCid)
+		case actorCodes[manifest.EamKey]:
+			acc.Require(actor.Head == emptyObjectCid, "Eam actor head %v unequal to emptyObjectCid %v", actor.Head, emptyObjectCid)
+		default:
+			return xerrors.Errorf("unexpected actor code CID %v for address %v", actor.Code, key)
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	// Check if all delegated addresses are part of init actor
+	for _, addr := range delegatedAddrs {
+		_, found := initSummary.AddrIDs[addr]
+		acc.Require(!found, "delegated address %v not found in init actor map", addr)
```

Comment:
```suggestion
		acc.Require(found, "delegated address %v not found in init actor map", addr)
```

Unless I'm an idiot?

Reply:
Nope. I'm the idiot here